pkg/config/spec.go

```diff
@@ -237,13 +250,26 @@ func NewUpgradeSpec(cfg *Config) (*v1.UpgradeSpec, error) {
 		}
 	}
 
-	return &v1.UpgradeSpec{
+	spec := &v1.UpgradeSpec{
 		Active:     active,
 		Recovery:   recovery,
 		Passive:    passive,
 		Partitions: ep,
 		State:      installState,
-	}, nil
+	}
+
+	// Get the actual source size to calculate the image size and partitions size
+	size, err := GetSourceSize(cfg, spec.Active.Source)
+	if err != nil {
+		cfg.Logger.Warnf("Failed to infer size for images: %s", err.Error())
+	} else {
+		cfg.Logger.Infof("Setting image size to %dMb", size)
+		spec.Active.Size = uint(size)
+		spec.Passive.Size = uint(size)
```

Comment:
I may be missing something here. When we upgrade, the passive image is the old one right? Could it be that the old one and the new one have different sizes? Is it ok to infer the size of the passive from the source?

Reply:
Passive is overwritten with the current active, so mainly a rename of active.img to passive.img and this value is ignored and only the active one is used.

So yeah, this is actually ignored and not used. Maybe I should infer the active image size and use that for passive? Before this, the default size was shown, so it's mainly esthetic, to show the values in the dump of the spec.

I will change it so it gets the default constant as it should not be override by this, it should have the default values, even if not used.